src/org/loklak/objects/Timeline2.java

```diff
@@ -0,0 +1,457 @@
+        JSONObject json = toSusi(withEnrichedData, new SusiThought(metadata_field_name, statuses_field_name));
+        json.getJSONObject(metadata_field_name).put("count", Integer.toString(this.posts.size()));
+        json.put("peer_hash", DAO.public_settings.getPeerHash());
+        json.put("peer_hash_algorithm", DAO.public_settings.getPeerHashAlgorithm());
+        return json;
+    }
+
+    public JSONObject toJSON() throws JSONException {
+        SusiThought susiThought = new SusiThought();
+        SusiThought json = toSusi(false, susiThought);
+        json.getJSONObject(susiThought.metadata_name).put("count", Integer.toString(this.posts.size()));
+        json.put("peer_hash", DAO.public_settings.getPeerHash());
+        json.put("peer_hash_algorithm", DAO.public_settings.getPeerHashAlgorithm());
+        return json;
+    }
+    
+    public SusiThought toSusi(boolean withEnrichedData) throws JSONException {
+        return toSusi(withEnrichedData, new SusiThought());
+    }
+    
+    private SusiThought toSusi(boolean withEnrichedData, SusiThought json) throws JSONException {
+        UserEntry user;
+        json.setQuery(this.query)
+            .setHits(Math.max(this.hits, this.size()));
+        if (this.scraperInfo.length() > 0) json.setScraperInfo(this.scraperInfo);
+        JSONArray statuses = new JSONArray();
+        for (JSONObject t: this) {
+            if(t.has("ScreenName")) {
+                user = this.users.get(t.get("ScreenName"));
+                // add user
+                if (user != null) t.put("user", user.toJSON());
+            }
+            statuses.put(t);
+            //TODO: add data with enriched data
+        }
+        json.setData(statuses);
+        return json;
+    }
+    
+    /**
+     * the tweet iterator returns tweets in descending appearance order (top first)
+     */
+    @Override
+    public Iterator<Post> iterator() {
+        return this.posts.descendingMap().values().iterator();
+    }
+
+    /**
+     * compute the average time between any two consecutive tweets
+     * @return time in milliseconds
+     */
+    public long period() {
+        if (this.size() < 1) return Long.MAX_VALUE;
+        
+        // calculate the time based on the latest 20 tweets (or less)
+        long latest = 0;
+        long time;
+        long earliest = 0;
+        int count = 0;
+        for (Post post: this) {
+            time = post.getTimestamp();
+            if (latest == 0) {
+                latest = time;
+                continue;
+            }
+            earliest = time;
+            count++;
+            if (count >= 19) break;
+        }
+
+        if (count == 0) return Long.MAX_VALUE;
+        long timeInterval = latest - earliest;
+        long p = 1 + timeInterval / count;
+        return p < 4000 ? p / 4 + 3000 : p;
+    }    
+
+    //TODO: this passes Timeline as argument
+    public void writeToIndex() {
```

Comment:
That is what I am talking about. Once you get more scrapers into work, you'll need to generalise the class for them. One of the requirement for it would be defining the index in which the data would need to get pushed.